File: tests/integration-tests/src/test/java/com/jaynewstrom/json/sample/required/RequiredTest.java
package com.jaynewstrom.json.sample.required;

import com.jaynewstrom.json.sample.JsonTestHelper;

import org.junit.Test;

import static org.fest.assertions.api.Assertions.assertThat;
import static org.junit.Assert.fail;

public final class RequiredTest {
    @Test public void whenRequiredFieldIsNull_ensureNullPointerIsThrown() {
        try {
            new JsonTestHelper().deserializeFile(Required.class, "RequiredTest.json", this);
            fail();
        } catch (NullPointerException e) {
            assertThat(e).hasMessage("foo == null");
        }
    }

    @Test public void whenRequiredFieldIsMissing_ensureNullPointerIsThrown() {
        try {
            new JsonTestHelper().deserializeFile(Required.class, "MissingTest.json", this);
            fail();
        } catch (NullPointerException e) {
            assertThat(e).hasMessage("foo == null");
        }
    }
}

File: tests/integration-tests/src/test/java/com/jaynewstrom/json/sample/required/RequiredTest.kt
package com.jaynewstrom.json.sample.required

import com.jaynewstrom.json.sample.JsonTestHelper
import org.fest.assertions.api.Assertions.assertThat
import org.junit.Assert.fail
import org.junit.Test

class RequiredTest {
    @Test fun whenRequiredFieldIsNull_ensureNullPointerIsThrown() {
        try {
            JsonTestHelper().deserializeFile(Required::class.java, "RequiredTest.json", this)
            fail()
        } catch (e: UninitializedPropertyAccessException) {
            assertThat(e).hasMessage("lateinit property foo has not been initialized")
        }
    }

    @Test fun whenRequiredFieldIsMissing_ensureNullPointerIsThrown() {
        try {
            JsonTestHelper().deserializeFile(Required::class.java, "MissingTest.json", this)
            fail()
        } catch (e: UninitializedPropertyAccessException) {
            assertThat(e).hasMessage("lateinit property foo has not been initialized")
        }
    }
}
